crates/sui-graphql-rpc/src/extensions/query_limits_checker.rs

```diff
@@ -326,8 +360,16 @@ impl QueryLimitsChecker {
     }
 }
 
-fn check_limits(limits: &Limits, cost: &ComponentCost, pos: Option<Pos>) -> ServerResult<()> {
+fn check_limits(
+    limits: &Limits,
+    cost: &ComponentCost,
+    pos: Option<Pos>,
+    ctx: &ExtensionContext<'_>,
```

Comment:
Can we pass in just the metrics instead of the entire extension context

Reply:
That makes sense!

Reply:
We need this to get the `query id` and the `session_id`